AWSAuthSDK/Sources/AWSMobileClient/Internal/AWSCognitoAuth+Extensions.h

```diff
@@ -19,6 +19,23 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface AWSCognitoAuthConfiguration(Extension)
 
+- (instancetype)initWithAppClientId:(NSString *) appClientId
+                    appClientSecret:(nullable NSString *)appClientSecret
+                             scopes:(NSSet<NSString *> *) scopes
+                  signInRedirectUri:(NSString *) signInRedirectUri
+                 signOutRedirectUri:(NSString *) signOutRedirectUri
+                          webDomain:(NSString *) webDomain
+                   identityProvider:(nullable NSString *)identityProvider
+                      idpIdentifier:(nullable NSString *)idpIdentifier
+                          signInUri:(nullable NSString *) signInUri
+                         signOutUri:(nullable NSString *) signOutUri
+                          tokensUri:(nullable NSString *) tokensUri
+           signInUriQueryParameters:(nullable NSDictionary<NSString *, NSString *> *) signInUriQueryParameters
+          signOutUriQueryParameters:(nullable NSDictionary<NSString *, NSString *> *) signOutUriQueryParameters
+            tokenUriQueryParameters:(nullable NSDictionary<NSString *, NSString *> *) tokenUriQueryParameters
+       userPoolServiceConfiguration:(nullable AWSServiceConfiguration *)serviceConfiguration
+               signInPrivateSession:(BOOL)signInPrivateSession;
+
```

Comment:
Why do we need both this and the old signature? If it's an internal API, can't we "just" add the new parameter and make sure it's appropriately set at the call sites? Or are there enough call sites that it would be cumbersome to do that?

Reply:
Removed the old API